### DISCUSSION TITLE
Zepp OS does support create BLE Beacon, based on watch?

### DISCUSSION THREAD

**Discussion by arstagaev** [zepp-health/zeppos-forum#307] [Q&A] [answered]
For example, in Wear OS I can create BLE Beacon / send advertise BLE packets every 1 sec. And Zepp OS support this? Or something like this 🙂?

https://source.android.com/docs/core/connect/bluetooth/ble_advertising

**Comment by silver-zepp** [comment]
`Zepp app -> Your watch -> Discoverable -> On`

Is that what you're looking for? @arstagaev

**Comment by arstagaev** [comment]
Thank you for answer @silver-zepp!

No, its mean what other devices can find Amazfit by BLE scanning. And I looking for BLE Beacon / BLE advertising - its when device send advertise packets.
If Zepp OS does not support BLE Beacon / BLE advertising - so I have a question :) I can modify BLE profile of my Amazfit? For sending some payload of bytes (count of steps per day and etc) to some other devices (not only to paired phone) **without connecting**

**Reply by silver-zepp** [reply, marked as answer]
Alright, was making sure we are on the same page. We currently don't support BLE Slave, so you will not be able to craft your own advertisement packets.

**Reply by arstagaev** [reply]
@silver-zepp Okay, I got it. And last two questions 😀, 

1. when I will setup state discoverable (Zepp app -> Your watch -> Discoverable -> On), Amazfit will be still discoverable, even it be connected to paired phone? (For example Samsung Galaxy Watch in connected state - becomes undiscoverable)
2.  I have another ways to send fitness data to my server? For example: Sending data to paired phone to special app -> and app will send data to server.
Or Amazfit has special WEB API, which can give access to fitness data from watches?

**Reply by silver-zepp** [reply]
Sure mate.

1. Yes, when discovery is `on`, you should be able to make multiple connections and parse generic data like FW/HW strings, battery level, etc, outside of a paired smartphone. But you will not be able to pair with it to get the detailed data you're looking for.
2. Our `REST API` is currently limited to big names, so the best route here would be to create your own `mini-app` that will parse the health data and send it to your server. If your device runs on `zos v3` you can also leverage the [Background Service (app-service)](https://docs.zepp.com/docs/guides/framework/device/app-service/) to seamlessly collect the data in the background. In case your watch runs on a lower generation fo zos, you can create a watch-face that will essentially do the same.

**Reply by arstagaev** [reply]
Thank you!
You mean `mini-app` based on Amazfit? And this `mini-app` can request some server?

**Reply by silver-zepp** [reply]
Yep, that's right. You can create your own apps in JavaScript and run them on your watch (assuming your watch runs any ZeppOS) You can check the list of the devices and their zos versions here: https://docs.zepp.com/docs/reference/related-resources/device-list
Here's a short intro into the development: https://youtu.be/RPyTge0kjWQ
Then you can proceed with communications, using a Fetch API: https://docs.zepp.com/docs/reference/side-service-api/fetch
Or ZML library: https://github.com/zepp-health/zml/
For a quick GUI creation, you can use this: https://github.com/silver-zepp/zeppos-autogui

And you should be good to go!

**Comment by arstagaev** [comment]
Hi @silver-zepp 🙂, any updates about ability to craft my own advertisement packets in Zepp OS or in Amazfit watches? Also interesting, I may customize (maybe even little part) current advertisement packets?

**Reply by arstagaev** [reply]
Hi @MyoungXUE 🙂, do you know answer to the question above?
